packages/cli/src/wiring/plugins/types.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CliFeature } from '../features/types';
+import { ServiceRef } from '../services/types';
+
+/** @ignore */
+type DepsToInstances<
+  TDeps extends {
+    [key in string]: ServiceRef<unknown>;
+  },
+> = {
+  [key in keyof TDeps]: TDeps[key] extends ServiceRef<unknown>
+    ? TDeps[key]['T']
+    : never;
+};
+
+/**
+ * The callbacks passed to the `register` method of a backend plugin.
+ *
+ * @public
+ */
+export interface CliPluginRegistrationPoints {
+  registerInit<
```

Comment:
Probably gonna want `registerExtensionPoint` too eventually right? For example, dynamically adding templates to the `new` command. That implies having modules as well as plugins, as a concept, of course.

Reply:
Likely so, wanted to keep the initial pass small